WcaOnRails/app/javascript/edit-events/EditEvents.jsx

```diff
@@ -0,0 +1,245 @@
+import React from 'react'
+import cn from 'classnames'
+import ReactDOM from 'react-dom'
+
+import events from 'wca/events.js.erb'
+import { rootRender, promiseSaveWcif } from 'edit-events'
+import { EditTimeLimitButton, EditCutoffButton, EditAdvancementConditionButton } from 'edit-events/modals'
+
+export default class EditEvents extends React.Component {
+  constructor(props) {
+    super(props);
+    this.save = this.save.bind(this);
+    this.onUnload = this.onUnload.bind(this);
+  }
+
+  save(e) {
+    let {competitionId, wcifEvents} = this.props;
+    let wcif = {
+      id: competitionId,
+      events: wcifEvents,
+    };
+
+    this.setState({ saving: true });
+    promiseSaveWcif(wcif).then(response => {
+      return response.json().then(json => [ response, json ]);
+    }).then(([response, json]) => {
```

Comment:
I like the use of `Promise.all` here!

Not sure if you also meant to make a point about indentation here, but I do not really like indenting my chained `then`s, because it looks weird when the chain ends and the level of indentation is more than 1 level off from the next one. See [here](https://github.com/beautify-web/js-beautify/issues/482#issuecomment-57046756) for a discussion including some folks who also have my school of thought on this.

I also don't want this PR to devolve into a discussion about javascript code formatting. I would like for us to get a linter and style checker set up, and then we can discuss how we want that thing to be configured. I've created #1924 to continue this discussion.